woocommerce/woocommerce-ios
# [Mobile Payments] Prevent flickering of the collect payment button on order details

Closes: #6780


### Description


We have decided to remove the check for whether there is a CPP eligible account or not when we show the Collect Payment button.

The check is now be handled by the onboarding flow, and removing the reliance on this results controller prevents the flickering we used to see on this button.

### Testing instructions


Payment-eligible orders:

1. Open an existing order which is pending payment (or processing) and card-payment eligible
2. Observe that the `Collect Payment` button is visible consistently from the moment the screen loads. Previously it flickered between `Issue Refund` and `Collect Payment`

Paid orders:
1. Open an existing order which is paid
2. Observe that the Issue Refund button is visible consistently from the moment the screen loads

### Screenshots

Before | After
---|---
![Flickering button before change](https://user-images.githubusercontent.com/2472348/172179366-05460207-8ca6-4b03-a2ae-40d5aa1a5741.mp4) | ![no flickering after change](https://user-images.githubusercontent.com/2472348/172179394-4c633023-3667-4b2e-80db-847cb569c2e8.mp4)


---
